systems/subscriber/sim-manager/pkg/server/sim_manager.go

```diff
@@ -229,6 +236,28 @@ func (s *SimManagerServer) AllocateSim(ctx context.Context, req *pb.AllocateSimR
 	if err != nil {
 		log.Errorf("Failed to publish message %+v with key %+v. Errors %s", req, route, err.Error())
 	}
+	netInfo, err := s.networkClient.GetNetworkInfo(remoteSubResp.Subscriber.NetworkId, remoteSubResp.Subscriber.OrgId)
```

Comment:
Is that remote call necessary?  A valid subscriber cannot exist  on an invalid network, so when we already have a valid subscriber, I think we can assume that we have a valid network for that subscriber (if the subscrier-registry has done well its job). So I don't think it is necessary to RPC validate the network again.

Reply:
To get the network details, we need to make a remote call to network. We don't have the network owner's name in the network database, so we call the network to get the network name for use in the email.